### ISSUE TITLE
Invalid hint for record field `id`

### ISSUE THREAD

**Issue by t4ccer** [ndmitchell/hlint#1383]
When using `OverloadedRecordDot` extension after defining record with field name `id`, hlint will give incorrect hints when field `id` of record is used via dot syntax.
### Example
```haskell
data Foo = Foo
  { id :: Int
  }

main = do
  let x = Foo 42
  print x.id
```
### Actual
`hlint src/Lib.hs`:
```
src/Lib.hs:9:3-12: Warning: Redundant id
Found:
  print x . id
Perhaps:
  print x
```
### Expected
No hints found on `print x.id` line
### Setup
`GHC`: `9.2.1`
`hlint:` `v3.4`/`052eff7` (`master` branch as of today)

**Comment by ndmitchell** [comment]
Thanks - as it stands HLint doesn't know about the OverloadedRecordDot extension, so thinks you have written a function composition. Given #1376 is incoming, let's wait and then once that lands, write a test for this so it doesn't happen again.

**Comment by svobot** [comment]
I don't know if this is useful, but another example that I've encountered is the following:
```haskell
data Foo = Foo
  { id :: Int
  }

main = print $ (Foo 42).id
```
Results in
```
Suggestion: Redundant bracket
Found:
  (Foo 42) . id
Perhaps:
  Foo 42 . id
```
hint.

**Comment by StevenXL** [comment]
@ndmitchell thank you very much for this library. I was having the issue as well, but it turned out it was because `OverloadedRecordSyntax` was not enabled at the (at the package or module level). Once that was enabled, things work. 

TLDR, should this issue be closed?

`hlint --version => HLint v3.5, (C) Neil Mitchell 2006-2022`.

**Comment by sirrus233** [comment]
I still get this issue on HLint v3.8, is there remaining work that needs to be done here? Or might there have been a regression?

**Comment by ncaq** [comment]
My HLint v3.8, I have problem too.

**Comment by ulidtko** [comment]
Also hit this on `v3.10`.

Root cause for me was misalignment of the actual `language` + `default-extensions` in package.yaml VS `arguments` configured in `.hlint.yaml`:

```yaml
# HLint configuration file
# https://github.com/ndmitchell/hlint

# Specify additional command line arguments
- arguments:
    # set edition, `language: GHC2021`
    - "-XGHC2021"
    # default-extensions:
    - "-XExplicitNamespaces"
    - "-XDataKinds"
    - "-XDerivingStrategies"
    - "-XDerivingVia"
    - "-XLambdaCase"
    - "-XMultiWayIf"
    # ...
    - "-XOverloadedRecordDot" # <-- fix
    # ...
#-- Keep in sync with `default-extensions` and `language` in package.yaml.
```

HLint itself supports `OverloadedRecordDot` since about v3.5 if not earlier — but the user has to tell hlint when/where the extension is active.
